discovery-provider/alembic/trigger_sql/handle_user_balance_changes.sql

```diff
@@ -8,20 +8,20 @@ declare
 begin
   SELECT label, val into new_tier, new_tier_value
   FROM (
-    VALUES ('bronze', 10), ('silver', 100), ('gold', 10000), ('platinum', 100000)
+    VALUES ('bronze', 10::bigint), ('silver', 100::bigint), ('gold', 10000::bigint), ('platinum', 100000::bigint)
   ) as tier (label, val)
   WHERE
-    new.current_balance::bigint >= tier.val
+    substr(new.current_balance, 1, GREATEST(1, length(new.current_balance) - 18))::bigint >= tier.val
```

Comment:
I did this because we store the current_balance as a string instead of number in postgres. This is bc there are 18 significant digits in Eth AUDIO and the largest bigint value in postgres is 9223372036854775807 which would mean the largest amount we can store is 9 AUDIO

But with strings, it is hard to compare numeric values, so I truncate the number by the 18 sig  digs and then compare to see if it passes the threshold